tests/test/show/test.sh

```diff
@@ -42,7 +42,6 @@ rlJournalStart
         rlAssertGrep "framework beakerlib" $rlRun_LOG
         rlAssertGrep "manual false" $rlRun_LOG
         rlAssertGrep "require.*required-package" $rlRun_LOG
-        rlAssertGrep "require.*beakerlib" $rlRun_LOG
```

Comment:
I'm not so sure about this anymore: On one hand it is not part of Test's require so why to show it. 
On the other hand it will be put into gathered require because of framework. And user might be surprised why it get there (especially if such package is not available...)

So the question is: How to make information about 'injected framework require' visible to the user?

Reply:
I have another aspect of the requirement handling: I was looking into those "essential" requirements, rsync & co., and it seems pretty much every single requirement collected but `Step.requires()` is of this essential category, with the exception of `discover` step which adds also test requirements. `requires()` is literally documented as "requirements which need to be installed on the provisioned guest so that all enabled plugins of this step can be successfully executed". Not tests, but plugins. It looks like we already do have a method to collect essential requirements, we just add test requirements on top of them as well.

And seeing your very good point about improving the user experience feels very related, we suddenly do distinguish essential requirements (`rsync` & co.), test framework requirements, and test requirements, and all these deserve their label as well ("why is rsync needed?!? oh, I see, that's why..."). So I was wondering, whether we could add some kind of enum or simple label to our `Dependency` classes, and simply label them as we discover them, and then code listing them to user could display them together with requirements. Something like `glibc, gcc, binutils, beakerlib (test framework), /usr/bin/ausearch (avc check)`. Or different colors, or `*` and `%`...

So, how about an issue to add this kind of tagging of requirements, and start logging it when needed? I'd say it would make nice addition, although I'd rather do it in an extra PR after the dust settles as this one and https://github.com/teemtee/tmt/pull/2567 get merged.

Reply:
Nice idea with user presentation, +1 to have it as a separate issue.